srtcore/core.h

```diff
@@ -255,6 +255,28 @@ struct MetricOp<PacketMetric>
     }
 };
 
+struct SRT_SocketOptionObject
+{
+    struct SingleOption
+    {
+        uint16_t option;
+        uint16_t length;
+        unsigned char storage[1]; // NOTE: Variable length object!
+    };
+
+    std::vector<SingleOption*> options;
+
+    SRT_SocketOptionObject() {}
+
+    ~SRT_SocketOptionObject()
+    {
+        for (size_t i = 0; i < options.size(); ++i)
+            delete options[i];
+    }
```

Comment:
Given that `option[i]` is allocated as `new unsigned char[headersize + payload]`, isn't the same conversion needed instead of the simple `delete options[i]`?

Something like `delete [] reinterpret_cast<unsigned char*>(options[i])`.

Reply:
Right.

Reply:
Another thing is that I'm not exactly sure that an array of unsigned char will not generate BUS ERROR. But I guess not on a single object (arrays of odd size packed one after another could have an odd address I think, but this fortunately isn't the case).